### PR TITLE
ci: run stdlib tests on macos

### PR DESCRIPTION
**Pull request by aykevl**

### CODE REVIEW

**Comment by dgryski** [comment]
Probably blocked on https://github.com/tinygo-org/tinygo/pull/2184

**Comment by aykevl** [comment]
Hopefully this fixes the issue.

**Comment by dgryski** [comment]
This builds on my mac, but fails with
```
=== RUN   TestOutOfBoundsRead
panic: runtime error: slice out of range
```

Again, I don't know why Linux doesn't have the same issue.

**Comment by aykevl** [comment]
It could very well be that I made a mistake in my `syscall.Mmap` implementation for MacOS. I didn't test it.

**Comment by dgryski** [comment]
This is passing locally for me now and in CI.

File: src/syscall/syscall_libc.go [hunk 1]
```diff
@@ -80,6 +80,22 @@ func Getenv(key string) (value string, found bool) {
 	}
 }
 
+func Mmap(fd int, offset int64, length int, prot int, flags int) (data []byte, err error) {
+	addr := libc_mmap(nil, uintptr(length), int32(prot), int32(flags), int32(fd), uintptr(offset))
+	if addr == unsafe.Pointer(^uintptr(0)) {
+		return nil, getErrno()
+	}
+	return (*[30]byte)(addr)[:length], nil
```
Comment (aykevl) [inline src/syscall/syscall_libc.go]: Well there is my bug. I meant to write:

```suggestion
	return (*[1 << 30]byte)(addr)[:length], nil
```

**Review by dgryski — APPROVED** [review]
LGTM and CI passes.

**Comment by aykevl** [comment]
Thanks! Looks to me too :)

**Pull request merged**